understanding/20/name-role-value.html

```diff
@@ -14,33 +14,40 @@ <h2>Intent of Name, Role, Value</h2>
       
       
       <p>The intent of this Success Criterion is to ensure that Assistive Technologies (AT)
-         can gather information about, activate (or set) and keep up to date on the status of
+         can gather appropriate information about, activate (or set) and keep up to date on the status of
          user interface controls in the content.
       </p>
       
       <p>When standard controls from accessible technologies are used, this process is straightforward.
          If the user interface elements are used according to specification the conditions
-         of this provision will be met. (See examples  of Success Criterion 4.1.2 below)
+         of this provision will be met. (See examples of Success Criterion 4.1.2 below)
       </p>
       
       <p>If custom controls are created, however, or interface elements are programmed (in
          code or script) to have a different role and/or function than usual, then additional
-         measures need to be taken to ensure that the controls provide important information
+         measures need to be taken to ensure that the controls provide important and appropriate information
          to assistive technologies and allow themselves to be controlled by assistive technologies.
       </p>
+     
+      <p>What roles and states are appropriate to convey to assistive technology will depend
+         on what the control represents. Specifics about such information are defined by other
+         specifications, such as <a href="https://www.w3.org/TR/wai-aria/">WAI-ARIA</a>, or the
+         appropriate platform standards. Another factor to consider is whether there is sufficient
+         support in assistive technologies to convey the information as specified.
+      </p>
       
       <p>A particularly important state of a user interface control is whether or not it has
          focus. The focus state of a control can be programmatically determined, and notifications
          about change of focus are sent to user agents and assistive technology. Other examples
          of user interface control state are whether or not a checkbox or radio button has
          been selected, or whether or not a collapsible tree or list node is expanded or collapsed.
       </p>
-      
+     
       <div class="note">
          
          <p>Success Criterion 4.1.2 requires a programmatically determinable name for all user
-            interface components. Names may be visible or invisible. Occasionally, the name must
-            be visible, in which case it is identified as a label. Refer to the definition of
+            interface components. Names may be visible or invisible. Occasionally, the name needs
+            to be visible, in which case it is identified as a label. Refer to the definition of
```

Comment:
perhaps "in which case it is identified as a label as well as a name." ?

Reply:
resolving this for now, will capture this in a separate issue/PR